src/pool/srv_target.c

```diff
@@ -2219,6 +2219,8 @@ cont_discard_cb(daos_handle_t ih, vos_iter_entry_t *entry,
 
 put:
 	ds_cont_child_put(cont);
+	if (rc == 0)
+		rc = ds_cont_child_destroy(arg->tgt_discard->pool_uuid, entry->ie_couuid);
```

Comment:
If it's a temporary workaround, it's better to have a comment to explain it, otherwise, reader could be confused why don't we destroy container directly (without discard).

Reply:
If the container is destroyed here, I think the snapshot IV fetch on rebuild scan handler will fail, right?